Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift

```diff
@@ -0,0 +1,379 @@
+//
+//  PrimerTestPaymentMethodTokenizationViewModel.swift
+//  PrimerSDK
+//
+//  Created by Dario Carlomagno on 25/05/22.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+class PrimerTestPaymentMethodTokenizationViewModel: PaymentMethodTokenizationViewModel {
+    
+    // MARK: - Properties
+    
+    internal private(set) var decisions = PrimerTestPaymentMethodOptions.FlowDecision.allCases
+    private var selectedDecision: PrimerTestPaymentMethodOptions.FlowDecision!
+    private var decisionSelectionCompletion: ((PrimerTestPaymentMethodOptions.FlowDecision) -> Void)?
+    private var payButtonTappedCompletion: (() -> Void)?
+    private var lastSelectedIndexPath: IndexPath?
+    
+    var viewHeight: CGFloat {
+        180+(CGFloat(decisions.count)*tableView.rowHeight)
+    }
+    
+    internal lazy var tableView: UITableView = {
+        let theme: PrimerThemeProtocol = DependencyContainer.resolve()
+        
+        let tableView = UITableView()
+        tableView.showsVerticalScrollIndicator = false
+        tableView.showsHorizontalScrollIndicator = false
+        tableView.rowHeight = 56
+        tableView.backgroundColor = theme.view.backgroundColor
+        if #available(iOS 11.0, *) {
+            tableView.contentInsetAdjustmentBehavior = .never
+        }        
+        tableView.register(FlowDecisionTableViewCell.self, forCellReuseIdentifier: FlowDecisionTableViewCell.identifier)
+        tableView.register(HeaderFooterLabelView.self, forHeaderFooterViewReuseIdentifier: "header")
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+    
+    private lazy var _originalImage: UIImage? = {
```

Comment:
Why don't you set them in `PaymentMethodTokenizationViewModel` as you did with Klarna?

Reply:
Good point. The reason why I decided to use the `originalImage` instead of the `buttonImage` is due to the fact that `originalImage` is utilized for the navigation bar logo in the flow decision view. 
The navigation bar has no background color there so I needed different variations of the PM logos.